libraries/opensk/src/ctap/data_formats.rs

```diff
@@ -748,6 +750,83 @@ impl CoseKey {
     const P_256_CURVE: i64 = 1;
     #[cfg(feature = "ed25519")]
     const ED25519_CURVE: i64 = 6;
+
+    // Ideally, this would be an impl From.
+    // CoseKey needs to be refactored to allow that.
```

Comment:
I'm not sure I follow why the problem is CoseKey. To me it looks like the problem is that we just can't have to `From<T> for CoseKey` implementations depending on which trait `T` implements. I think the current solution is the right one.

Reply:
I was wondering if Rust has a way to also have different `CoseKey<A: Algorithm>` and you get a `CoseKey<ECDSA>::from(EcdsaPublicKey)` and the compiler checks correctness because you never implemented `CoseKey<ECDSA>::from(EcdhPublicKey)`.

Then CoseKey also wouldn't have to store its other map keys, only `x_bytes` and `y_bytes`. (This optimization can also be achieved indepedently by having an enum property of course.)

But this is good enough for me, if there is nothing Rust-y I'm missing.

Reply:
Agreed to the last point. I think my question is closer to: Should we have a `EcdhCoseKey` and a `EcdsaCoseKey`? But this probably comes down to being an implementation detail or micro optimization that we shouldn't care about too much.

Reply:
I see, I don't know enough how those are used in CBOR, but if it's always clear whether we want to parse an `EcdhCoseKey` or an `EcdsaCoseKey` then this would work too indeed. Not sure if worth trying to do though.